server/vertx/src/main/scala/sttp/tapir/server/vertx/encoders/VertxOutputEncoders.scala

```diff
@@ -0,0 +1,173 @@
+package sttp.tapir.server.vertx.encoders
+
+import java.io.{File, InputStream}
+import java.nio.ByteBuffer
+import java.nio.charset.Charset
+
+import io.vertx.core.buffer.Buffer
+import io.vertx.core.http.HttpHeaders
+import io.vertx.scala.core.http.HttpServerResponse
+import io.vertx.scala.core.streams.ReadStream
+import io.vertx.scala.ext.web.RoutingContext
+import sttp.model.{Header, Part}
+import sttp.tapir.internal.ParamsAsAny
+import sttp.tapir.server.ServerDefaults
+import sttp.tapir.server.internal.{EncodeOutputBody, EncodeOutputs, OutputValues}
+import sttp.tapir.server.vertx.VertxEndpointOptions
+import sttp.tapir.{CodecFormat, Endpoint, EndpointOutput, RawBodyType}
+
+/**
+ * All the necessary methods to write Endpoint.outputs to Vert.x HttpServerResponse
+ * Contains:
+ *   - Headers handling
+ *   - Body handling (string, binaries, files, multipart, ...)
+ *   - Stream handling
+ */
+object VertxOutputEncoders {
+
+  type RoutingContextHandler = RoutingContext => Unit
+  type RoutingContextHandlerWithLength = Option[Long] => RoutingContext => Unit
+
+  /**
+   * Creates a function, that given a RoutingContext will write the result to its response, according to the endpoint definition
+   * @param output the endpoint definition
+   * @param result the result to encode
+   * @param isError boolean indicating if the result is an error or not
+   *
+   * @tparam O type of the result to encode
+   * @return a function, that given a RoutingContext will write the output to its HTTP response
+   */
+  private [vertx] def apply[O](output: EndpointOutput[O], result: O, isError: Boolean = false, log: Int => Unit = { _ => })
```

Comment:
maybe there could be a better name for `log`? It's meant to be logging what kind of data?

Reply:
Done (+ javadoc)